kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -362,6 +365,25 @@ protected void close() {
                 currentConnectedGroup.remove(clientHost);
             }
 
+            // Try to remove all stored groupID on the metadata store.
+            if (log.isDebugEnabled()) {
+                log.debug("Try to remove all stored groupID on the metadata store. Current connected clientIds: {}",
+                        currentConnectedClientId);
+            }
+            currentConnectedClientId.forEach(clientId -> {
+                String path = groupIdStoredPath + GroupIdUtils.groupIdPathFormat(clientHost, clientId);
+                metadataStore.deleteRecursive(path)
```

Comment:
IMO, it's better to use `delete` here because the path should not contain any child.

I think it would be better if we could separate all these z-node create/get/delete operations from `KafkaRequestHandler` and `MessageFetchContext` to a single class like `GroupIdStorage`.

Reply:
Yes, `delete` is better. We can open other PR to refactor to `GroupIdStorage `.